planner/core/logical_plan_test.go

```diff
@@ -1269,6 +1269,32 @@ func (s *testPlanSuite) TestColumnPruning(c *C) {
 	}
 }
 
+func (s *testPlanSuite) TestProjectionEliminater(c *C) {
+	defer testleak.AfterTest(c)()
+	tests := []struct {
+		sql  string
+		best string
+	}{
+		{
+			sql:  "select 1+num from (select 1+a as num from t) t1;",
+			best: "DataScan(t)->Projection->Projection",
```

Comment:
Still two projections here?

Reply:
I dont know why...
But I try this sql in terminal.The result shows the projection is really be eliminated.
```
Projection_4	4.00	root	plus(1, plus(1, test.t.a))
└─TableReader_6	4.00	root	data:TableScan_5
  └─TableScan_5	4.00	cop	table:t, range:[-inf,+inf], keep order:false, stats:pseudo
```